spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/CassandraPartitionGenerator.scala

```diff
@@ -50,10 +50,20 @@ private[connector] class CassandraPartitionGenerator[V, T <: Token[V]](
   }
 
   private def describeRing: Seq[TokenRange] = {
-    connector.withClusterDo { cluster =>
+    val ranges = connector.withClusterDo { cluster =>
       val metadata = cluster.getMetadata
       for (tr <- metadata.getTokenRanges.toSeq) yield tokenRange(tr, metadata)
     }
+
+    /**
+      * When we have a single Spark Partition use a single global range. This
+      * will let us more easily deal with Partition Key equals and In clauses
+      */
+    if (splitCount == Some(1)) {
+      Seq(ranges(0).copy[V, T](tokenFactory.minToken, tokenFactory.maxToken))
```

Comment:
It has valid replicas but invalid datasize. Basically if we only have a single partition it covers the entire token range and our replica code will pick only local dc replicas (or should at least prioritize them). I considered creating a new range from scratch but you still need to correctly grab hosts for replicas to ensure you are in the right datacenter. 

This ended up seeming like a much cleaner solution for the time being than rewriting the Range creation code.